mantidproject/mantid
# Add Ability to Load Existing Calibration into Engineering Diffraction UI

**Description of work.**
Adds the ability to load a GSAS parameter file into the engineering diffraction interface. This sets the current calibration of the interface internally and also sets up the interface to redo the creation of a calibration if required.

**To test:**
1. Test in Mantid plot & workbench
2. Open the interface via Interfaces -> Diffraction -> Engineering Diffraction 2
    a. You may have to add `Diffraction/Engineering_DIffraction_2.py` to the mantid.properties file. 
3. Generate a new calibration (V: 307521 Cal: 305738) if you do not have one already.
4. Click the load radio button and browse to the created file (`~/Engineering_Mantid/Calibration/something.prm`)
5. Check that the right values were loaded into the GUI, you can test this by clicking the create new radio button and running the calibration again. 

Fixes #26848 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
